### PR TITLE
libczmqcontainers: include zlistx, zhash, zlist, and convert internal users

### PR DESCRIPTION
**Pull request by garlick**
As noted in #3617, it may be beneficial to eliminate the `libflux-core.so` dependence on `libczmq.so` to avoid a FIPS self-check performance hit in the DSO constructor of `libcrypto.so` for all Flux API users including python modules and the flux-shell.

Since many users of czmq in flux-core are only using its containers, and we have already created the internal library `libczmqcontainers` to pull in a local copy of zhashx for a bug fix, this PR takes the next step and pulls in the other containers (zlistx, zhash, and zlist), makes libczmqcontainers code independent of libczmq, and visits most of the places where containers are used and points them to the internal library instead of the external `czmq.h`.

One material change is that `zmalloc()` (czmq's internal malloc function) no longer asserts on ENOMEM failure.  Other than that, there should be no functional changes in this PR.

### CODE REVIEW

File: src/common/libczmqcontainers/czmq_containers.h [hunk 1]
```diff
@@ -15,6 +15,15 @@
 extern "C" {
 #endif
 
+#include <czmq.h>
```
Comment (chu11) [inline src/common/libczmqcontainers/czmq_containers.h]: commit message "libcontainers" -> "libczmqcontainers"

**Review by chu11 — APPROVED** [review]
LGTM, pretty straight forward. Just one commit header typo I noticed.

**Comment by chu11** [comment]
> Define zmalloc() in terms of calloc().  This means that container functions that would assert on malloc error can now return failure. (Most of our code assumes this anyway).

I just realized, this probably isn't true in many cases.  Alot of the code in czmq is of this style.

```
    zhashx_t *self = (zhashx_t *) zmalloc (sizeof (zhashx_t));
    assert (self);
    self->prime_index = INITIAL_PRIME;
```

the result of a calloc error would likely lead to a segfault.  I had considered modifying the code in PR #3596 to return `NULL` (or appropriate return code) but decided against it as it'd be too many changes.

**Comment by garlick** [comment]
Thanks @chu11 - oh yeah, you are right.

I audited all the `zmalloc()` call sites and it appears that they all have an internal assertion check like your example above, except
* `zhashx_pack_own()` will de-reference a NULL pointer (but we don't build that function)
* `zhash_insert()` will now return  -1 if a hash resize fails, but not if the node insert fails

Since `zhash_update()` calls `zhash_insert()` without checking for failure, this introduces a bug.

I'll go ahead and restore the old behavior for now.  Possibly we could fix the above issues upstream and pull them down again, but given the general policy of checking `zalloc()` with `assert()` I'm not sure it would make much difference.

**Comment by garlick** [comment]
I force pushed the `zmalloc()` change, and also reverted changes to `librlist` to revisit later since I was running afoul of its FFI wrapper in the python bindings, and didn't want to spend more time on this right now to sort  it out.

**Review by SteVwonder — COMMENTED** [review]
Thanks @garlick .  Didn't give a full review, was just glancing at it and notice the comment below:

**Comment by garlick** [comment]
OK, forced a push with conversion of `librlist`, after an assist from @grondo on the python bindings.  I also pulled in his fix to the `make_clean_header.py` script, which was creating an output file on failure and thus confusing the build system.

File: t/module/parent.c [hunk 2]
```diff
@@ -17,10 +17,11 @@
 #include <libgen.h>
 #include <dlfcn.h>
 #include <argz.h>
-#include <flux/core.h>
 #include <czmq.h>
```
[the remark this answers is not in the document]
Comment (garlick) [reply t/module/parent.c:20]: Oh just saw this.  I think so!  Thanks, I'll amend one more time.

**Comment by garlick** [comment]
Another force push with the fix pointed out by @SteVwonder .

**Comment by garlick** [comment]
OK one more for the LGTM warning.  This time for sure!

**Comment by garlick** [comment]
I messed up my testing of that change to `t/module/parent.c` - turns out it needs czmq for another class.  Reverted.

**Comment by garlick** [comment]
Any objection if I set MWP on this?

**Review by grondo — APPROVED** [review]
LGTM! Feel free to set MWP when ready as far as I'm concerned.

**Comment by chu11** [comment]
@garlick LGTM too, MWP away

**Comment by grondo** [comment]
@mergifyio rebase

**Comment by grondo** [comment]
I hope I didn't break our mergify config with my latest change. I tried getting mergify to do a manual rebase. If that doesn't work we may have to back out or redo the latest update to `.mergify.yml`

**Pull request merged**